google-cloud-clients/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/models/QueryTest.java

```diff
@@ -227,4 +227,48 @@ public void shardTestKeyOffsets() {
         .setTableName(TABLE_NAME.toString())
         .setAppProfileId(APP_PROFILE_ID);
   }
+
+  @Test
+  public void testFromProto() {
+    ReadRowsRequest request = ReadRowsRequest.newBuilder()
+        .setTableName(TABLE_NAME.toString())
+        .setAppProfileId(APP_PROFILE_ID)
+        .setFilter(
+            RowFilter.newBuilder()
+                .setRowKeyRegexFilter(ByteString.copyFromUtf8(".*")))
+        .setRows(RowSet.newBuilder()
+                .addRowKeys(ByteString.copyFromUtf8("row-key"))
+                .addRowRanges(
+                    RowRange.newBuilder()
+                        .setStartKeyClosed(ByteString.copyFromUtf8("j"))
+                        .setEndKeyClosed(ByteString.copyFromUtf8("z"))))
+        .build();
+    Query query = Query.fromProto(request);
+
+    assertThat(query.toProto(requestContext))
+        .isEqualTo(request);
+  }
+
+  @Test
+  public void testFromProtoWithEmptyTableId() {
```

Comment:
This relates to the strategy of converting `Scan` to something serializable in Beam.  We need some sort of placeholder for `Table` until `ValueProvider`s materialize.  We were thinking about empty as that placeholder.  We cause an alternative placeholder string.

Reply:
Might be good to use something else as the placeholder. Now I've deprecated the typesafe names, there is a much higher risk of a user failing to specify an id. At some point soon, I would like to add validations for the names.

Reply:
Agree to use Placeholder instead of an empty string.
We need to use the placeholders at [CloudBigtableScanConfiguration#build](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java#L273) as well, So would like to have uniform Placeholders across the projects.

 Does `EMPTY_TABLE_ID` sound good? or should it be more specific?

Reply:
Yes, I would use this placeholder in [beam/CloudBigtableScanConfiguration.java](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java) & [beam/TemplateUtils.java](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java). Because in both of this place I would not have `tableId`.

Reply:
SGTM